python/pyiceberg/cli/console.py

```diff
@@ -242,6 +242,29 @@ def rename(ctx: Context, from_identifier: str, to_identifier: str) -> None:
     catalog.rename_table(from_identifier, to_identifier)
     output.text(f"Renamed table from {from_identifier} to {to_identifier}")
 
+@run.command()
+@click.argument("identifier")
+@click.option("--type", required=False)
+@click.option("--verbose", type=click.BOOL)
+@click.pass_context
+@catch_exception()
+def list_refs(ctx: Context, identifier: str, type: str, verbose: bool) -> None:
+    """List all the refs in the provided table"""
+    catalog, output = _catalog_and_output(ctx)
+    table = catalog.load_table(identifier)
+    refs = table.metadata.refs
+    if type:
+        type = type.lower()
+        if type != "branch" and type != "tag":
+            raise ValueError("Type must be either branch or tag")
+
+    relevant_refs = []
```

Comment:
A bit more pythonic, you can do this on a single line as well:
```suggestion
    relevant_refs = [
        ref_name, ref for ref_name, ref in refs.items() if not type or ref.snapshot_ref_type == type
    ]
```